### DISCUSSION TITLE
Fehler bei Bottle Framework installieren ‐ obsolet ab V2

### DISCUSSION THREAD

**Discussion by Ingo-LMC** [spitzlbergerj/CaravanPi#33] [Fragen und Antworten]
Hallo,
Ich hänge leider bei dem Punkt fest, der Rest hat bisher alles geklappt.
Wenn ich den Befehl laut Anleitung eingebe bekomme ich eine Fehlermeldung, siehe unten. 
Ich freue mich auf Hilfestellung!
LG Ingo

CaravanPi@CaravanPi:~ $
CaravanPi@CaravanPi:~ $
CaravanPi@CaravanPi:~ $ pip3 install bottle
error: externally-managed-environment

This environment is externally managed

To install Python packages system-wide, try apt install
python3-xyz, where xyz is the package you are trying to
install.

If you wish to install a non-Debian-packaged Python package,
create a virtual environment using python3 -m venv path/to/venv.
Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
sure you have python3-full installed.

For more information visit http://rptl.io/venv

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
CaravanPi@CaravanPi:~ $

**Comment by Ingo-LMC** [comment]
Ich habe weitergeschaut und nun einmal einen Step nach vorne gemacht: Ich habe des Installationsskript V2 einmal komplett nach Anleitung drüberlaufen lassen. Das hat auch soweit geklappt... nun stehe ich aber schon wieder am nächsten Punkt, nämlich das Problem mit dem Schalter:

CaravanPi@CaravanPi:~ $
CaravanPi@CaravanPi:~ $
CaravanPi@CaravanPi:~ $ sudo cp /home/pi/CaravanPi/pishutdown/pishutdown.py /usr/local/bin
sudo cp /home/pi/CaravanPi/pishutdown/pishutdown.service /etc/systemd/system
sudo systemctl enable pishutdown
sudo systemctl start pishutdown
cp: der Aufruf von stat f�r '/home/pi/CaravanPi/pishutdown/pishutdown.py' ist nicht m�glich: Datei oder Verzeichnis nicht gefunden
cp: der Aufruf von stat f�r '/home/pi/CaravanPi/pishutdown/pishutdown.service' ist nicht m�glich: Datei oder Verzeichnis nicht gefunden
Failed to enable unit: Unit file pishutdown.service does not exist.
Failed to start pishutdown.service: Unit pishutdown.service not found.
CaravanPi@CaravanPi:~ $

**Comment by Ingo-LMC** [comment]
Und: Bei der aktuellen Ansicht des Magic Mirros wird "nur" Datum und Uhrzeit korrekt angezeigt, bei den anderen Daten bleibt es bei Loading stehen... 
Wenn ich mir die Config anschaue: Fehlt da nicht das:

basePath: "/",	// The URL path where MagicMirror² is hosted. If you are using a Reverse proxy
									// you must set the sub path here. basePath must end with a /

Ich habe es schon mal probiert einzufügen, aber es ändert nichts... 

Ich freue mich auf euer Feedback!

LG Ingo

**Comment by spitzlbergerj** [comment]
servus Ingo,

wir sind grad unterwegs und ich habe keinen Rechner dabei. Daher kann ich nicht nachsehen.

das Installationsskript wäre mein erster Tipp gewesen. pip install funktioniert in neuen pi os versionen nicht mehr. Leider haben das die hersteller der archive bisher ignoriert. Das Skript schafft da Abhilfe.

pishutdown ist eine Kombi aus Software und Hardware. Das muss ich mir detaillierter ansehen.

MagicMirror hatte ich zuletzt auch Probleme. Ich muss mir die Lösung wieder raus suchen.

Ich kann mich frühestens Sonntag Abend melden.

Viele Grüße
Sepp
________________________________
Von: Ingo ***@***.***>
Gesendet: Samstag, 26. Oktober 2024 01:04
An: spitzlbergerj/CaravanPi
Cc: Subscribed
Betreff: Re: [spitzlbergerj/CaravanPi] Fehler bei Bottle Framework installieren ‐ obsolet ab V2 (Discussion #33)


Und: Bei der aktuellen Ansicht des Magic Mirros wird "nur" Datum und Uhrzeit korrekt angezeigt, bei den anderen Daten bleibt es bei Loading stehen...
Wenn ich mir die Config anschaue: Fehlt da nicht das:

basePath: "/", // The URL path where MagicMirror² is hosted. If you are using a Reverse proxy
// you must set the sub path here. basePath must end with a /

Ich habe es schon mal probiert einzufügen, aber es ändert nichts...

Ich freue mich auf euer Feedback!

LG Ingo

**Comment by Ingo-LMC** [comment]
Hallo Sepp,
Vielen Dank für Deine Antwort.
Ich habe gestern noch bis halb drei Nachts getüftelt und habe das Feedback des Herrn hier mir zu nutze gemacht: Ich habe auch einmal komplett neu gestartet und die Benutzernamen auf pi belassen, den hatte ich nämlich vorher auch auf CaravanPI (genau wie der Hostname) geändert. Danach habe ich dann direkt nur das Skript ausgeführt und hier hat alles geklappt. 
An dem angeschlossenen Monitor konnte ich auch zugucken und sehen das MM am Anfang incl. Wetter Daten geklappt hat und nachdem das Script durchgelaufen ist steht wie gesagt nur Loading da. Anbei ein Screenshot.

Nochmals vielen Dank für Deine Mithilfe, ich bin richtig angfixt und freue mich so mal wieder in die Programmierwelt und Raspberry PI Welt einzutauchen. 
![CaravanPI_MM_Ohne Wetter](https://github.com/user-attachments/assets/ee53a0bc-5edf-4ebb-ac5d-6b25a9463b95)

**Comment by spitzlbergerj** [comment]
Servus Ingo,

ich komme grad nicht dazu, die Platine einzupacken. ich bin leider sehr beschäftigt. ich werde das aber im Lauf der Woche fertig machen und versenden. Danke auch für den zu hohen Betrag. Das freut mich sehr.

Schau die doch bitte mal die config des MagicMirrors an. Ich meine, da sind in der config von github nur ein paar Beispiel Module drin, die nicht richtig konfiguriert sind. Suche Dir die Module, die du wirklich am Spiegel haben willst und konfiguriere die dann gemäß Anleitung auf github zum jeweiligen Modul. Dann sollte das klappen.

Wenn du dann die Platine hast und Sensoren arbeiten, zeige ich Dir, wie du das Grafana Modul konfigurierst.

Sorry für die Verzögerung, aber ich kann leider noch nicht "nur basteln" .... 🙂

Viele Grüße

Sepp


________________________________
Von: Ingo ***@***.***>
Gesendet: Samstag, 26. Oktober 2024 15:33
An: spitzlbergerj/CaravanPi ***@***.***>
Cc: Josef Spitzlberger ***@***.***>; Comment ***@***.***>
Betreff: Re: [spitzlbergerj/CaravanPi] Fehler bei Bottle Framework installieren ‐ obsolet ab V2 (Discussion #33)


Hallo Sepp,
Vielen Dank für Deine Antwort.
Ich habe gestern noch bis halb drei Nachts getüftelt und habe das Feedback des Herrn hier mir zu nutze gemacht: Ich habe auch einmal komplett neu gestartet und die Benutzernamen auf pi belassen, den hatte ich nämlich vorher auch auf CaravanPI (genau wie der Hostname) geändert. Danach habe ich dann direkt nur das Skript ausgeführt und hier hat alles geklappt.
An dem angeschlossenen Monitor konnte ich auch zugucken und sehen das MM am Anfang incl. Wetter Daten geklappt hat und nachdem das Script durchgelaufen ist steht wie gesagt nur Loading da. Anbei ein Screenshot.

Nochmals vielen Dank für Deine Mithilfe, ich bin richtig angfixt und freue mich so mal wieder in die Programmierwelt und Raspberry PI Welt einzutauchen.
CaravanPI_MM_Ohne.Wetter.png (view on web)<https://github.com/user-attachments/assets/ee53a0bc-5edf-4ebb-ac5d-6b25a9463b95>

**Reply by Ingo-LMC** [reply]
Hallo Sepp,

Vielen Dank für Deine Antwort. Mach Dir bitte keinen Stress... ich habe gestern den Wohnwagen ins Winterquartier gefahren und somit brennt hier nichts an. Ich habe also Zeit alles über den Winter im Büro aufzubauen und zu testen :-)

Die Temperatur Sensoren und Klmiasensoren (welche für mich erstmal am wichtigsten sind) habe ich bereits bestellt und wenn die Platine dann da ist kann ich den "Trocken-Aufbau" in Ruhe machen und alles testen... dann werde ich DIch bestimmt nochmal "belästigen" wenn Du erlaubst :-)

Beste Grüße vom Niederrhein und nochmals Danke für Deine tolle Hilfe!

VG Ingo

**Comment by Ingo-LMC** [comment]
Leider bekomme ich immer noch nicht die Wetter Daten ans Laufen. Den API Key habe ich auch eingetragen. Ich verstehe es nicht.... 
Wenn jemand einen Tip hat dann freue ich micht, hier ein paar Screenshots meiner Analyse.

    {
      module: 'weather',
      position: 'top_right',
      header: 'Wetteraussichten',
      config: {
        type: 'forecast',
        roundTemp: true,
        colored: true,
        showPrecipitationAmount: true,
        fade: false,
        locationID: '2820465',
        apiKey: 'c7a560749a983cacda9ffd42aa93781e',
        maxNumberOfDays: 5
      }
    },


![grafik](https://github.com/user-attachments/assets/32733ea5-aec8-4029-a072-13a3f2cf4453)

**Comment by spitzlbergerj** [comment]
Hallo Ingo,
Du hast noch alle möglichen Module aktiv, die Fehler werfen. Kommentiere erst mal alles aus bis auf die Zeit und das Logo. Dann aktiviere Modul für Modul und suche jeweils die Fehler.

bitte schicke mir doch an meine Mailadresse deine config für den MagicMirror. Ich schau mir das an.

Grüße
Sepp
